platform/osgi/platform-osgi-conditions/src/test/java/org/codice/ddf/condition/BundleNameConditionTest.java

```diff
@@ -0,0 +1,140 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.condition;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.service.condpermadmin.Condition;
+import org.osgi.service.condpermadmin.ConditionInfo;
+
+public class BundleNameConditionTest {
+
+  @Test
+  public void testIsSatisfied() {
+    Bundle bundle = mock(Bundle.class);
+    Dictionary<String, String> headers = new Hashtable<>();
+    headers.put("Bundle-SymbolicName", "bundle2");
+    when(bundle.getHeaders()).thenReturn(headers);
+    BundleNameCondition bundleNameCondition =
+        new BundleNameCondition(
+            bundle,
+            new ConditionInfo(
+                BundleNameCondition.class.getName(), new String[] {"bundle1/bundle2/bundle3"}));
+    boolean satisfied = bundleNameCondition.isSatisfied();
+    // no cache this time around
+    assertThat(satisfied, is(true));
+    satisfied = bundleNameCondition.isSatisfied();
+    // cached
+    assertThat(satisfied, is(true));
+  }
+
+  @Test
+  public void testIsNotSatisfied() {
+    Bundle bundle = mock(Bundle.class);
+    Dictionary<String, String> headers = new Hashtable<>();
+    headers.put("Bundle-SymbolicName", "bundle4");
+    when(bundle.getHeaders()).thenReturn(headers);
+    BundleNameCondition bundleNameCondition =
+        new BundleNameCondition(
+            bundle,
+            new ConditionInfo(
+                BundleNameCondition.class.getName(), new String[] {"bundle1/bundle2/bundle3"}));
+    boolean satisfied = bundleNameCondition.isSatisfied();
+    // no cache this time around
+    assertThat(satisfied, is(false));
+    satisfied = bundleNameCondition.isSatisfied();
+    // cached
+    assertThat(satisfied, is(false));
+  }
+
+  @Test
+  public void testIsNotSatisfiedNoArgs() {
+    Bundle bundle = mock(Bundle.class);
+    Dictionary<String, String> headers = new Hashtable<>();
+    headers.put("Bundle-SymbolicName", "bundle4");
+    when(bundle.getHeaders()).thenReturn(headers);
+    BundleNameCondition bundleNameCondition =
+        new BundleNameCondition(
+            bundle, new ConditionInfo(BundleNameCondition.class.getName(), new String[0]));
+    boolean satisfied = bundleNameCondition.isSatisfied();
+    // no cache this time around
+    assertThat(satisfied, is(false));
+    satisfied = bundleNameCondition.isSatisfied();
+    // cached
+    assertThat(satisfied, is(false));
+  }
+
+  @Test
+  public void testIsAllSatisfied() {
```

Comment:
❓ Since `isSatisfied(Condition[], Dictionary)` is defined in `AbstractCondition` and doesn't really care about the actual `Condition` implementation, should we put those tests in a separate `Test` class instead of making them look like they're `BundleNameCondition` specific?

Reply:
I could I guess, or I could put the same test in all of them, or I can just leave it as is. I don't really feel strongly about it either way as long as the code is tested.